ngraph_bridge/ngraph_builder.cc

```diff
@@ -4825,6 +4825,55 @@ static Status TranslateUnpackOp(const Node* op,
   return Status::OK();
 }
 
+static Status TranslateUnsortedSegmentSumOp(
+    const Node* op, const std::vector<const Tensor*>& static_input_map,
+    Builder::OpMap& ng_op_map) {
+  /*
+  # Code from python that reflects how unsorted_segment_sum
+  # works in tensorflow using scatter_add
+
+  def unsorted_segment_sum(inputs, segment_ids, num_segments):
+    inputs = tf.convert_to_tensor(inputs)
+    segment_ids = tf.convert_to_tensor(segment_ids)
+
+    shape = inputs.get_shape().as_list()
+    seg_ids = segment_ids.get_shape().as_list()
+    result = tf.Variable(tf.zeros([num_segments] + shape[len(seg_ids):],
+                                  dtype=inputs.dtype))
+
+    return tf.scatter_add(result, segment_ids, inputs)
+  */
+
+  shared_ptr<ng::Node> ng_input, ng_segment_ids, ng_num_segments;
+  TF_RETURN_IF_ERROR(GetInputNodes(ng_op_map, op, &ng_input, &ng_segment_ids,
+                                   &ng_num_segments));
```

Comment:
Input number 2 is actually read in by `GetStaticInputVector` in the next line, so we dont need to read in `ng_num_segments` here.

We can pass:
```
TF_RETURN_IF_ERROR(GetInputNodes(ng_op_map, op, &ng_input, &ng_segment_ids, nullptr));
```
to indicate that there are 3 inputs, of which get the first 2, but we dont care about the third

Reply:
Fixed, thanks.